docs/Diagnostics.md

```diff
@@ -150,3 +150,5 @@ An expected diagnostic is denoted by a comment which begins with `expected-error
 - (Required) The expected error message. The message should be enclosed in double curly braces and should not include the `error:`/`warning:`/`note:`/`remark:` prefix. For example, `// expected-error {{invalid redeclaration of 'y'}}` would match an error with that message on the same line. The expected message does not need to match the emitted message verbatim. As long as the expected message is a substring of the original message, they will match.
 
 - (Optional) Expected fix-its. These are each enclosed in double curly braces and appear after the expected message. An expected fix-it consists of a column range followed by the text it's expected to be replaced with. For example, `let r : Int i = j // expected-error{{consecutive statements}} {{12-12=;}}` will match a fix-it attached to the consecutive statements error which inserts a semicolon at column 12, just after the 't' in 'Int'. The special {{none}} specifier is also supported, which will cause the diagnostic match to fail if unexpected fix-its are produced.
+
+- (Optional) Expected educational notes. These appear as a comma separated list after the expected message, enclosed in double curly braces and prefixed by 'educational-notes='. For example, `{{educational-notes=some-note,some-other-note}}` will verify the educational notes with filenames `some-note` and `some-other-note` appear. Do not include the file extension when specifying note names. The special `{{educational-notes=none}}` specifier is also supported, which will cause the diagnostic match to fail if any unexpected educational notes are produced.
```

Comment:
Perhaps we should use a syntax more akin to the other diagnostic selectors.

```
// expected-educational-notes {{one,two,three}} 
```

Reply:
I'm also unsure of the utility of `{{educational-notes=none}}`. Testing for their absence seems like an anti-pattern.

Reply:
I prefer the current syntax because it's clear the notes are attached to a specific diagnostic, just like fix-its, and don't have source location information of their own. I did drop the 'none' specifier though, as I haven't needed it yet in practice.

Reply:
Ah, I see what the intent is here then. I was under the impression that educational notes were more like notes than fixits.

Reply:
Yeah, the distinction between a ”note” and an “educational note” is kind of odd from a compiler-internal perspective, but it makes it really easy to explain the concept to end users so the name stuck. I should document the difference at some point.